### PR TITLE
Checkbox: use id passed in if provided

### LINKED ISSUE #6081: Checkbox does not honor id property
When you create a checkbox and specify the id attribute, it is ignored:

This:
```
<Checkbox id="ignoredId" 
        label="Checkbox with ignored Id" 
        onChange={this._onCheckboxChange} 
        ariaDescribedBy={'descriptionID'} />
```
Produces:
```
<button id="checkbox-1" role="checkbox" type="button" class="ms-Checkbox is-enabled root-47" aria-checked="false" aria-describedby="descriptionID">
    <label class="ms-Checkbox-label label-48" for="checkbox-1">
        <div class="ms-Checkbox-checkbox checkbox-49">
            <i data-icon-name="CheckMark" class="ms-Checkbox-checkmark checkmark-53" role="presentation" aria-hidden="true"></i>
        </div>
        <span class="ms-Checkbox-text text-51">Standard 2 checkbox</span>
    </label>
</button>
```

The id is "checkbox-1" instead of "ignoredId".

I'm trying to automate some texts with selenium and cannot get the checkboxes by id.

Here is a repro: https://codepen.io/Intnovaction/pen/xaZvPN

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #6081 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Use the id passed in for the checkbox if it is provided.

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Focus should be on the selected item. I rejected the screener test because the description was "Multi-select: click", and the interactive state was removed, but now I realize that was a focus state. I went back and accepted.

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.test.tsx [hunk 1]
```diff
@@ -9,4 +9,11 @@ describe('Checkbox', () => {
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
+  it('renders Checkbox with id correctly', () => {
+    const component = renderer.create(
+      <Checkbox label="Standard checkbox" ariaDescribedBy={'descriptionID'} id="my-checkbox" />
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.test.tsx]: can we just test the ID of the node? a whole snapshot is a lot of noise if we ever make changes to checkbox that don't affect the ID at all.

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx [hunk 2]
```diff
@@ -94,7 +95,7 @@ export class CheckboxBase extends BaseComponent<ICheckboxProps, ICheckboxState>
             disabled={disabled}
             ref={this._checkBox}
             name={name}
-            id={this._id}
+            id={id ? id : this._id}
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx]: pretty sure this._id is used lateron down in the component. Can we just set this._id to either the generated value, or the passed in one? That way we don't have to keep checking for ID when we want the default ID value.

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export class CheckboxBase extends BaseComponent<ICheckboxProps, ICheckboxState>
       checked: 'defaultChecked'
     });
 
-    this._id = getId('checkbox-');
+    this._id = this.props.id ? this.props.id : getId('checkbox-');
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx]: could simplify with this.props.id || getId('checkbox-);

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.test.tsx [hunk 4]
```diff
@@ -10,10 +11,7 @@ describe('Checkbox', () => {
     expect(tree).toMatchSnapshot();
   });
   it('renders Checkbox with id correctly', () => {
-    const component = renderer.create(
-      <Checkbox label="Standard checkbox" ariaDescribedBy={'descriptionID'} id="my-checkbox" />
-    );
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    const component = mount(<Checkbox label="Standard checkbox" ariaDescribedBy={'descriptionID'} id="my-checkbox" />);
+    expect(component.find('button').prop('id')).toEqual('my-checkbox');
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.test.tsx]: awesome

**Pull request merged**